### PR TITLE
chore: update to Vaadin 24

### PR DESCRIPTION
**Pull request by ZheSun88**

### CODE REVIEW

File: vaadin-archetype-application/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -42,9 +42,9 @@
         </dependency>
 
         <dependency>
-            <groupId>javax.servlet</groupId>
-            <artifactId>javax.servlet-api</artifactId>
-            <version>4.0.1</version>
+            <groupId>jakarta.servlet</groupId>
+            <artifactId>jakarta.servlet-api</artifactId>
+            <version>5.0.0</version>
```
Comment (mshabarov) [inline vaadin-archetype-application/src/main/resources/archetype-resources/pom.xml:47]:
```suggestion
            <version>6.0.0</version>
```

**Review by mshabarov — DISMISSED** [review]

File: vaadin-archetype-application/src/main/resources/archetype-resources/pom.xml (hunk 1 above)
Comment (ZheSun88) [reply vaadin-archetype-application/src/main/resources/archetype-resources/pom.xml]: done

File: vaadin-archetype-application/src/main/resources/archetype-resources/pom.xml [hunk 2]
```diff
@@ -55,7 +55,7 @@
             <plugin>
                 <groupId>org.eclipse.jetty</groupId>
                 <artifactId>jetty-maven-plugin</artifactId>
-                <version>10.0.13</version>
+                <version>11.0.13</version>
```
Comment (mstahv) [inline vaadin-archetype-application/src/main/resources/archetype-resources/pom.xml:58]:
```suggestion
                <version>11.0.14</version>
```

Comment (mstahv) [reply vaadin-archetype-application/src/main/resources/archetype-resources/pom.xml:58]: There is a newer jetty plugin available

**Review by mstahv — CHANGES_REQUESTED** [review]

**Review by mstahv — APPROVED** [review]
Smoke tested locally both archetypes, built against 24.0.0, all seems to be fine!

**Pull request merged**